### ISSUE TITLE
ImageClassifier1D stuck after 3 models

### ISSUE THREAD

**Issue by zhan2977** [keras-team/autokeras#642]
### Bug Description

ImageClassifier1D gets stuck after training 3 models.

### Reproducing Steps
```
import autokeras
from autokeras.image.image_supervised import ImageClassifier1D

import numpy as np
import os
import pickle

os.environ["CUDA_VISIBLE_DEVICES"]=str(2)

def shuffle(x_data, seed=195735):
    shuffled_x = np.empty(x_data.shape, dtype=x_data.dtype)
    
    for class_index in range(15):
        for snr_index in range(21):
            permutation = np.random.permutation(x_data.shape[2])
            for old_index, new_index in enumerate(permutation):
                shuffled_x[class_index,snr_index,new_index,:,:] = x_data[class_index,snr_index,old_index,:,:]
    
    return shuffled_x

def split_data(x_data, y_data):
    
    # Shuffle data
    x_data = shuffle(x_data)
    
    # Split data
    x_train = x_data[:,:,:572,:,:]
    y_train = y_data[:,:,:572]
    x_test = x_data[:,:,572:,:,:]
    y_test = y_data[:,:,572:]
    
    return x_train, y_train, x_test, y_test

def load_data(storage_folder, which_kind):

    x_data = None
    y_data = None
    data_file = None
    # decide which data to load
    if which_kind == 'iq':
        data_file = 'data_iq.p'
    elif which_kind == 'fft':
        data_file = 'data_fft.p'
    elif which_kind == 'amplitude phase':
        data_file = 'data_amplitude_phase.p'
    elif which_kind == 'fft amplitude phase':
        data_file = 'data_fft_amplitude_phase.p'
    else:
        raise ValueError('Parameter which_kind must be "iq" for IQ-data or "fft" for FFT-data.')
    # load input data (x)
    data_path = os.path.join(storage_folder, data_file)
    with open(data_path, mode='rb') as storage: 
        x_data = pickle.load(storage)
    # load output data/labels (y)
    label_file = 'labels.p'
    label_path = os.path.join(storage_folder, label_file)
    with open(label_path, mode='rb') as storage:
        y_data = pickle.load(storage, encoding='latin1')
    return x_data, y_data

def normalize_data(x_data):
    mean = np.mean(x_data, axis=(0,1,2))
    std_dev = np.std(x_data, axis=(0,1,2))    
    x_data = (x_data - mean) / std_dev
    return x_data

folder = 'data'
x_data, y_data = load_data(folder, 'fft')
x_data = normalize_data(x_data)

x_train, y_train, x_test, y_test = split_data(x_data, y_data)

x_train = x_train.reshape(15*21*572, 128, 2)
x_test = x_test.reshape(15*21*143, 128, 2)

y_train = y_train.reshape(15*21*572)
y_test = y_test.reshape(15*21*143)

clf = ImageClassifier1D(verbose=True)
clf.fit(x_train, y_train, time_limit=12*60*60)
```

### Additional context

Saving Directory: /tmp/autokeras_K4R56C
Preprocessing the images.
Preprocessing finished.

Initializing search.
Initialization finished.

+----------------------------------------------+
|               Training model 0               |
+----------------------------------------------+
                                                                                                    
No loss decrease after 5 epochs.

Saving model.
+--------------------------------------------------------------------------+
|        Model ID        |          Loss          |      Metric Value      |
+--------------------------------------------------------------------------+
|           0            |   4.930431747436524    |         0.5708         |
+--------------------------------------------------------------------------+

+----------------------------------------------+
|               Training model 1               |
+----------------------------------------------+
                                                                                                    
No loss decrease after 5 epochs.

Saving model.
+--------------------------------------------------------------------------+
|        Model ID        |          Loss          |      Metric Value      |
+--------------------------------------------------------------------------+
|           1            |   1.6132919669151307   |   0.8775999999999999   |
+--------------------------------------------------------------------------+

+----------------------------------------------+
|               Training model 2               |
+----------------------------------------------+
                                                                                                    
No loss decrease after 5 epochs.

Saving model.
+--------------------------------------------------------------------------+
|        Model ID        |          Loss          |      Metric Value      |
+--------------------------------------------------------------------------+
|           2            |   1.3866336643695831   |   0.8795999999999999   |
+--------------------------------------------------------------------------+

+----------------------------------------------+
|               Training model 3               |
+----------------------------------------------+
                                                                                                    
No loss decrease after 5 epochs.

**Comment by devanshkv** [comment]
@zhan2977 I am also getting the same issue.
